### DISCUSSION TITLE
Ads?

### DISCUSSION THREAD

**Discussion by pietrop** [vidstack/player#703] [Ideas]
_non urgent question_

Was wondering if there's any thoughts on how to handle pre-roll ads? Eg google ads?

Eg whether this would be done outside of the player via composition (eg in React)? Or whether the player would also have components for this?

**Comment by mihar-22** [comment]
Honestly, I haven't implemented ads before so not sure. I don't know the inner workings to comment yet. We could unpack this together with your guidance (if of any interest).

**Reply by pietrop** [reply]
Sounds good, thanks!

**Comment by Zaied** [comment]
Any update on ads support? Like Google IMA SDK?
https://developers.google.com/interactive-media-ads/docs/sdks/html5/client-side/compatibility

**Comment by connecteev** [comment]
@mihar-22 @pietrop not urgent, but I am also interested / curious about how to do this using Vidstack (ie Pre-roll, mid-roll skippable ads, post-roll ads)

**Comment by aokigit** [comment]
Also very interested in ad support. Only thing that's preventing me from using this player. 😭

**Comment by yohannmarchand** [comment]
Any news on this feature request ?

**Comment by ft-scobra** [comment]
Non urgent. Would also like to see this implemented in the future. It would give me an option to add an ad supported tier for my website.
